### DISCUSSION TITLE
Interpreting science-technology pair as need-solution pair

### DISCUSSION THREAD

**Discussion by hyunjimoon** [Data4DM/BayesSD#169] [people; relating]
<img width="1241" alt="image" src="https://github.com/Data4DM/BayesSD/assets/30194633/1c60e854-29c6-4a85-a87e-1f15a2a8795a">

1. Macro economic model of two stocks with the knowledge of science and technology are as below; causal model in the left and stock and flow diagram in the right (labor of scientist and technologist can be stock as well). I wonder whether the reference mode of science-technology would be closer to that of prey-predator or to that of actual-measured room temperature dynamics. Some analysis on feedback loops and delay:
- science-technology is `3R0B` when all six policy parameters are positive. However, a model without any balancing loop is less realistic. It may be more practical to allow signs of $\phi, \psi$ to shift between negative (burden of knowldge, fishing out) and positive (recombination) to curb the exponential growth. Exponential growth is not supported empirical evidence (??). prey-predator is `2R3B` (or `2B2R` when birth and death rates are not influenced by the population is constant e.g. enough food and space in the environment. `3B2R` means when there is more population they make more babies and die more e.g. enough food but limited space); actual-measured temperature is `0R2B`. Among the eight system architecture, science-technology seems to resemble power dynamics / success to successful most.
<img width="835" alt="image" src="https://github.com/Data4DM/BayesSD/assets/30194633/bd18bdac-432e-4337-8920-70f6f57c0970"> 

- Material and mental delay have different structure in that input and output are usually modeled separately in the former.  Tweaking the proportion of each six parameters would have equivalent effect of tweaking adjustment time and measurement delay for room temperature model and tweaking birth and death fraction in  prey and predator model.

Process noise can be injected to represent unknown/unrepresented uncertainty. I still don't have clear understanding on how process noise in the generator (which we may be able find corresponding hierarchical generator e.g. hyper parameter sampled from half normal distribution) affects estimation of hierarchical model.
<img width="1271" alt="image" src="https://github.com/Data4DM/BayesSD/assets/30194633/b1580c50-5055-4372-8f1e-419ac94af3c5">

4. I wonder whether redrawing below (reference room temperature) on the axis of adjustment fraction (inverse of parameter) may make reference modes more cluster-able. It would be a hyperbolic space
<img width="664" alt="image" src="https://github.com/Data4DM/BayesSD/assets/30194633/9bc23f1f-6dc5-4d14-9fd9-ac430fa0723d">

**Comment by tomfid** [comment]
Regarding #2 (the plot from BATS), I've been wondering whether the modes here are really distinct. OSCCT and GR2DB probably differ in damping. But is SSHGR even more damped so it can't overshoot, or is it just that the time scale gets longer as the delay gets longer, so the peak in GR2DB ends too late in the model time horizon to be observed?

**Reply by hyunjimoon** [reply]
So it all depends on timestep and time horizon as they are measured by correlation -- let me link the code in a min!

**Reply by hyunjimoon** [reply]
https://github.com/hyunjimoon/DataInDM/blob/9adbd4eb93a51f7b951ae64a12d9009128562c46/Code_Commit/4_Auto/GeomPrior/aria/test_scripts/ists_test.py#L59

**Reply by hyunjimoon** [reply]
Chat with Yaman which may help us understanding:

Angie:
1. ists_instance.smooth(y) -> ists_instance.smooth(y, alpha): I am not sure which value I should give to alpha.
2. ists.covariance(y) -> ists.covariance(y, lag):adding lag for covariance is needed.

Yaman:
Thank you for the corrections.
- Alpha you mention is the alpha value used in standard expon smoothing. It is up to the user, typically between 0.01 and 0.20. Alpha is equivalent to 1/AT used in delay formulations (smoothing). (In time-continuous models, it is like (1/AT)*DT). Small alpha means strong smoothing of noisy data. Large alpha means 
very little smoothing (the output nearly mimicking the noisy input 
data).

You are right, autocovariance lag is a required user input (ists would compute the autocorrelations up to that lag).

**Comment by hyunjimoon** [comment]
vensim files for three models: 
[twostock_uc.zip](https://github.com/Data4DM/BayesSD/files/13054307/twostock_uc.zip)

**Comment by tomfid** [comment]
Regarding the tech model, I like it. We had some similar components in models for DOE Office of Science a few years ago. This is a nice compact representation.

I don't think the temp control model is a good analog though. They're topologically similar, but the gains and nonlinearities are different. The temp control loops are negative, while the tech model is dominated by positive feedback.

**Reply by hyunjimoon** [reply]
Yes, I felt tech model is a little unrealistic in that it doesn't have any balance loop.

**Reply by tomfid** [reply]
I'm not sure what you had in mind for the loop through "sci2" but I think the sign of theta is an interesting question. There might be an effect of ideas spawning further ideas (positive theta) but I think the dominant effect is that problems get incrementally harder (negative theta) making it a balancing loop. There appears to be some evidence for this. https://web.stanford.edu/~chadj/IdeaPF.pdf

**Reply by tomfid** [reply]
I think the tech2 loop is also mixed. Better tools definitely help develop better tools, e.g., you can use a computer to CAD a better screwdriver, and a better screwdriver to make a better computer. This is reinforcing. But then it also gets harder to improve the computer or screwdriver incrementally, for a fixed supply of B. 

I think a modification might be needed to capture the idea that progress on A depends on how much of B you have already exploited. For example, if you discover the thermoelectric effect (a B item), you can then make Peltier coolers (an A item). But once you've done that, you can only make small improvements, until you get some new B to exploit.

![image](https://github.com/Data4DM/BayesSD/assets/2151849/f6af4c4d-8a15-49cb-8bfa-17147b3771cb)

**Reply by hyunjimoon** [reply]
Oh I see, this model stems from a pset for a class I'm taking below. Scott seems to be very interested in the tech2sci feedback. 
[15_357_Pset1_QA.pdf](https://github.com/Data4DM/BayesSD/files/13074583/15_357_Pset1_QA.pdf)

**Reply by hyunjimoon** [reply]
Including the following paper which estimated tech2sci by collecting the data of science organization that couldn't use physical implementation/ experience due to catastrophe (I think , didn't read too carefully)
[Baruffaldi22_Returns2PhysicalCapital4KnowledgeProduction.pdf](https://github.com/Data4DM/BayesSD/files/13074678/Baruffaldi22_Returns2PhysicalCapital4KnowledgeProduction.pdf)

**Reply by tomfid** [reply]
It's possible that the "unexploited principles" loop is already implicit in the equation for dA. You could rewrite it as:

dA/dt = A^a * B^b * (B/A)^c * L^k
phi = a-c
lambda = b+c

I think the main implication is that 'a' (phi in the tech2 equation on the diagram) doesn't have a clean interpretation, as it mixes several effects.

**Reply by hyunjimoon** [reply]
>  'a' (phi in the tech2 equation on the diagram) doesn't have a clean interpretation, as it mixes several effects.

Might "mixed" be similar to "residual" from [Solow's residual](https://www.investopedia.com/terms/s/solow-residual.asp)?

**Reply by tomfid** [reply]
In the sense that Solow's residual means "stuff other than capital and labor that we attribute to tech" you might say the same about the roles of A&B here.

Incidentally people usually think of the Solow tech as computers, software, etc. but it's also soft tech, like better market instruments or contracts, as well as human capital.

**Comment by hyunjimoon** [comment]
One of the most insightful comment from today's talk by Adam Jaffe was his comment that data is public good and is by definition non-rivorous and excludable but through data-basing efforts like [I³ Tracker for Essential Open Innovation Datasets](https://github.com/Innovation-Information-Initiative/Open-Innovation-Dataset-Index#i-tracker-for-essential-open-innovation-datasets) + [spreadsheet](https://docs.google.com/spreadsheets/d/1bdyhGrj0oNz-_qW3Rv2GNGqhZZ73rgj-DYWePLA_1Ms/edit?usp=sharing), Adam is trying to push data from excludable to non-excludable.

**Comment by hyunjimoon** [comment]
synthesizing three diagrams (angie-tom's diagram + the last two from this thread) my conclusion is we need measurement 🦁beast!  <img width="1475" alt="image" src="https://github.com/user-attachments/assets/778d2b0a-8bca-40a2-8f6e-23a56ba7f55e">

---

@tomfid's [tom under theory implem.mdl.zip](https://github.com/user-attachments/files/18082860/tom.under.theory.implem.mdl.zip) contains one important question that needs to be answered: how does the entrepreneur know which direction to change an attribute in order to improve "quality". tom gave the example of sharper knife better than duller one as simple quality measure but argued in most cases quality measures are multidimensional and ambiguous. Good example is vensim - i have no idea whether target customer would want cleaner graphical interface with a few functions, or a complicated one with lots of functions. But this is a function of target customer which itself is a function of vision. vision comes from quality of imaginative understanding.

Focusing on measurement, ecosystem innovation policy makers should provide a good measurement infrastructure/guideline (analogy is for each entrepreneur's compass to work well, underlying magnetic field should be increasingly measured hence partially controlled).

Tom also shared his confusion on form and function, like @jeanbaptiste. Thinking of the knife example, form is function, but I don't think that's what you mean. Does this relate to product vs. process innovation, or ... ?

![image](https://github.com/user-attachments/assets/c28c031c-0819-4dd8-88c8-3339d7886db0)

---
> <img alt="image" width="1241" src="https://private-user-images.githubusercontent.com/30194633/276929035-1c60e854-29c6-4a85-a87e-1f15a2a8795a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4NDg0MjQsIm5iZiI6MTczMzg0ODEyNCwicGF0aCI6Ii8zMDE5NDYzMy8yNzY5MjkwMzUtMWM2MGU4NTQtMjljNi00YTg1LWE4N2UtMWYxNWEyYTg3OTVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEwVDE2Mjg0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk2NTBlOGYzMzhlY2NkMmVkZTM3ZjJhMzUyM2JiMjc3OGM5NGY3NmZiMTdkY2MwMDc0Y2VkYjc1NWJjMzhmZGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.puWgwmTMxfraYwJGY_0J1Duogq_2qQK8QwDHshIVH4E">

 <img alt="image" width="1271" src="https://private-user-images.githubusercontent.com/30194633/276931429-b1580c50-5055-4372-8f1e-419ac94af3c5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4NDg0MjQsIm5iZiI6MTczMzg0ODEyNCwicGF0aCI6Ii8zMDE5NDYzMy8yNzY5MzE0MjktYjE1ODBjNTAtNTA1NS00MzcyLThmMWUtNDE5YWM5NGFmM2M1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEwVDE2Mjg0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVkOTkzNDU3N2JkZGI0ZmQ1MDVjMjdkNzliMWY1NGU3ZmUyZDkzOTcyNDZhOTYzYTFhZWI3MjdiMGMwMzA2OTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1TSqMR2JNvSQ8jPDVebUVME_QmYN5n4jlWbl6XZ4Fq4">
